examples/ceed/ex2_surface.py

```diff
@@ -0,0 +1,212 @@
+
+def get_cartesian_mesh_size(dim, degree, prob_size):
+    num_elem = prob_size // (degree ** dim)
+    s = 0
+    while 2**s <= num_elem:
+        s += 1
+    s -= 1
+    r = s % dim
+    num_xyz = []
+    for d in range(dim):
+        sd = s // dim
+        if r > 0:
+            sd += 1
+            r -= 1
+        num_xyz.append(1 << sd)
+    return num_xyz
+
+def build_cartesian_restriction(ceed, dim, num_xyz, degree, num_comp, num_qpts):
+    p = degree + 1
+    num_nodes = p ** dim
+    elem_qpts = num_qpts ** dim
+    nd = []
+    num_elem = 1
+    scalar_size = 1
+    for d in range(dim):
+        num_elem *= num_xyz[d]
+        nd.append(num_xyz[d] * (p - 1) + 1)
+        scalar_size *= nd[d]
+    size = scalar_size * num_comp
+    elem_nodes = np.zeros(num_elem * num_nodes, dtype=np.int32)
+    for e in range(num_elem):
+        e_xyz = [0] * dim
+        re = e
+        for d in range(dim):
+            e_xyz[d] = re % num_xyz[d]
+            re //= num_xyz[d]
+        for n in range(num_nodes):
+            g_node = 0
+            g_stride = 1
+            r_node = n
+            for d in range(dim):
+                g_node += (e_xyz[d] * (p - 1) + r_node % p) * g_stride
+                g_stride *= nd[d]
+                r_node //= p
+            elem_nodes[e * num_nodes + n] = g_node
+    elem_restriction = ceed.ElemRestriction(
+        num_elem, num_nodes, num_comp, scalar_size, size, elem_nodes)
+    qd_comp = dim * (dim + 1) // 2
+    strides = np.array([1, elem_qpts, elem_qpts * qd_comp], dtype=np.int32)
+    q_data_restriction = ceed.StridedElemRestriction(
+        num_elem, elem_qpts, qd_comp, num_elem * elem_qpts * qd_comp, strides)
+    return elem_restriction, size, q_data_restriction, num_elem, elem_qpts
+
+def set_cartesian_mesh_coords(ceed, dim, num_xyz, mesh_degree, mesh_coords):
+    p = mesh_degree + 1
+    nd = []
+    scalar_size = 1
+    for d in range(dim):
+        nd.append(num_xyz[d] * (p - 1) + 1)
+        scalar_size *= nd[d]
+    coords = np.zeros(scalar_size * dim)
+    nodes, _ = ceed.lobatto_quadrature(p)
+    nodes = 0.5 + 0.5 * nodes
+    for gs_node in range(scalar_size):
+        r_node = gs_node
+        for d in range(dim):
+            d_1d = r_node % nd[d]
+            elem_id = d_1d // (p - 1)
+            node_id = d_1d % (p - 1)
+            coords[gs_node + scalar_size * d] = (elem_id + nodes[node_id]) / num_xyz[d]
+            r_node //= nd[d]
+    mesh_coords.set_array(coords, cmode=libceed.COPY_VALUES)
+
+def transform_mesh_coords(dim, mesh_size, mesh_coords):
+    exact_measure = {1: 2.0, 2: 4.0, 3: 6.0}[dim]
+    with mesh_coords.array() as coords:
+        for d in range(dim):
+            offset = d * (mesh_size // dim)
+            for i in range(mesh_size // dim):
+                coords[offset + i] = 0.5 + (1.0/np.sqrt(3.0)) * np.sin(
+                    (2.0/3.0) * np.pi * (coords[offset + i] - 0.5))
+    return exact_measure
+
+def main():
+    args = parse_arguments()
```

Comment:
```suggestion
def example_2(args):
```
See the discussion here: https://github.com/CEED/libCEED/issues/1811

You'll want a `main` below this fn with the contents
```py
def main():
    """Main function for surface area computation example"""
    args = parse_arguments()
    return example_2(args)
```

Reply:
I’ve reviewed the discussion and updated my code to follow the structure used in the volume example, including adjusting object usage and function layout. However, I’m still having trouble with formatting.
I ran make format-py, and while it shows formatting changes in the diff, those changes aren't being committed to the branch. Could you help me ensure that the formatted code is properly staged and committed?